pymc3/step_methods/metropolis.py

```diff
@@ -616,6 +616,145 @@ def competence(var, has_grad):
         return Competence.COMPATIBLE
 
 
+class DEMetropolisZ(ArrayStepShared):
```

Comment:
Could we use a more informative name?

Reply:
Suggestions?
But in its behavior it's similar to `DEMetropolis` in many ways. For both algorithms the user would be well-advised to read the paper anyway.

Reply:
`DEHistoryMetropolis`?

Reply:
The name used in the paper is DE-MCz. Given that we already have a class named DEMetropolis, DEMetropolisZ sounds like a good compromise to me.

Reply:
Cool, let's go with DEMetropolisZ then.